Zend/zend_API.c

```diff
@@ -558,7 +558,7 @@ static const char *zend_parse_arg_impl(int arg_num, zval *arg, va_list *va, cons
 				zval **p = va_arg(*va, zval **);
 
 				if (!zend_parse_arg_number(arg, p, check_null)) {
-					return "number";
+					return check_null ? "number or null" : "number";
```

Comment:
This should be `int|float` and `int|float|null`. (FastZPP already uses `int|float`.)